auth.go

```diff
@@ -396,3 +396,52 @@ func prepareJWTToken(config *Config) (string, error) {
 
 	return tokenString, err
 }
+
+//authenticate with sc.cfg
```

Comment:
Looks like this method:
1. call `authenticateByExternalBrowser` or `authenticateBySaml`
2. then call `authenticate`
3. then set the auth response session information into the connection's session param and context

What do you think about calling this `authenticateSession(...)` or `initializeSession`?

```
// Authenticate and initialize the the session parameters
func authenticateSession(sc *snowflakeConn) error { ... }
```

Reply:
> Looks like this method:
> 
> 1. call `authenticateByExternalBrowser` or `authenticateBySaml`
> 2. then call `authenticate`
> 3. then set the auth response session information into the connection's session param and context
> 
> What do you think about calling this `authenticateSession(...)` or `initializeSession`?
> The comment can also be better formatted and more descriptive
> 
> ```
> // Authenticate and initialize the the session parameters
> func authenticateSession(sc *snowflakeConn) error { ... }
> ```

This method is doing authenticate with config in sc.cfg, so the user of this function will know they could provide config in sc.cfg to call this method to do authentication. Also I feel that initSession is too general.